src/App/App.js

```diff
@@ -0,0 +1,202 @@
+import React, { Component } from 'react';
+import Login from '../Login/Login';
+import MovieGrid from '../MovieGrid/MovieGrid';
+import MovieDetails from '../MovieDetails/MovieDetails';
+import Search from '../Search/Search';
+import Banner from '../Banner/Banner';
+import { fetchMovies, fetchSingleMovie, postLoginCredentials } from '../apiCalls';
+import { Route, NavLink } from 'react-router-dom';
+import DropdownButton from 'react-bootstrap/DropdownButton';
+import './App.css';
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      movies: [],
+      error: '',
+      currentUser: null,
+      input: '',
+      dropdownValue: 'all',
+      ratingValue: 'any'
+    }
+  }
+  
+  componentDidMount = () => {
+    fetchMovies()
+    .then(allMovies => this.setState({ movies: allMovies.movies}))
+    .then(() => this.findMovieGenres())
+    .catch(error => this.setState({ error: error.message }))
+  }
+
+  findMovieGenres = () => {
+    this.state.movies.map(movie => {
+      fetchSingleMovie(movie.id)
+      .then(singleMovie => movie.genres = singleMovie.movie.genres)
+    })
+  }
+
+  validateLogin = (loginEmail, loginPassword) => {
+    const credentials = {
+      email: loginEmail,
+      password: loginPassword
+    }
+
+    postLoginCredentials(credentials)
+    .then(user => this.setState({ currentUser: user.user }))
+    .catch(error => this.setState({ error: error.message}))
+  }
+
+  clearError = () => {
+    this.setState({ error: '' })
+  }
+
+  signOut = () => {
+    this.setState({ currentUser: null })
+  }
+
+  get filterMoviesBySelection() {
+    const filterByGenre = this.state.movies.filter(movie => {
+      if (this.state.dropdownValue !== 'all' && movie.genres) {
+        return movie.genres.includes(this.state.dropdownValue)
+      } else {
+        return this.state.movies
+      }
+    })
+
+    const filterByRating = filterByGenre.filter(movie => {
+      if (this.state.ratingValue !== 'any') {
+        const roundedRating = parseInt(movie.average_rating)
+        return roundedRating === parseInt(this.state.ratingValue)
+      } else {
+        return filterByGenre
+      }
+    })
+
+    return filterByRating.filter(movie => {
+      return movie.title.toLowerCase().includes(this.state.input)
+    })
+  }
+
+  getUserInput = (inputValue) => {
+    this.setState({input: inputValue})
+  }
+
+  handleDropdownValue = (event) => {
+    this.setState({ dropdownValue: event.target.name})
+  }
+
+  handleRatingValue = (event) => {
+    this.setState({ ratingValue: event.target.value })
+  }
+
+  clearFilters = () => {
+    this.setState({ input: '', dropdownValue: 'all', ratingValue: 'any' })
+  }
+
+  render() {
+    return(
+      <main>
+        <header>
+          <h1 className='page-title'>underrated</h1>
```

Comment:
I was expecting this to be a link to the index page because it is where the logo is.